lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2550,6 +2573,65 @@ void ModuleEmitter::emitStatementBlock(Block &body) {
 // Module Driver
 //===----------------------------------------------------------------------===//
 
+void SplitModuleEmitter::emitMLIRModule(ModuleOp module) {
+  for (auto &op : *module.getBody()) {
+    // TODO: Module uniquification happens in ModuleEmitter, but a new instance
+    // of ModuleEmitter is created for each operation in the body. We could add
+    // uniquification across all operations by moving the renaming table into a
+    // separate state object that is passed around, like a VerilogEmitterState.
```

Comment:
I think renaming might have to be a prepass.  It needs to be consistent regardless of what subset of the file the emitter is looking at.

Reply:
The filenames need to reflect renaming, so I think it has to be early.